### DISCUSSION TITLE
Is it garanteed that `ImageBuffer::new()` returns a buffer filled with `zero()`s of the corresponding type ?

### DISCUSSION THREAD

**Discussion by Morgane55440** [image-rs/image#2217] [Q&A] [answered]
i'm working on a feature in [`imageproc`](https://github.com/image-rs/imageproc/issues/576), and for optimization reasons, i would like to rely on the fact that `ImageBuffer::new()` returns a buffer filled with `zero()`s.

but [the documentation](https://docs.rs/image/0.25.1/image/struct.ImageBuffer.html#method.new) of `ImageBuffer::new()` only states

> Creates a new image buffer based on a Vec<P::Subpixel>.
Panics
Panics when the resulting image is larger than the maximum size of a vector

Therefore, i am looking for confirmation outside of the current implementation that this is an intended feature that will most likely not be changed in the future rather than an implementation detail that could be changed for any and all reasons at any moment.

**Comment by 197g** [comment, marked as answer]
Yes, I don't see any reason to change that and it is usually the most efficient initialization routine there is. We can agree on this behavior without evolution issue.

**Reply by fintelia** [reply]
Agreed. I'd welcome a PR adding this guarantee to the documentation

**Reply by Morgane55440** [reply]
i have just made such a pull request : #2219
